### PR TITLE
Provide utils to dump component configuration as json string

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

*Motivation*

It is useful to dump configuration during service components started.
Bookie server already did that.

*Changes*

- Move JsonUtil to bookkeeper-common module to make it available to other modules.
- Add toJson method to ComponentConfiguration

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [ ] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

File: bookkeeper-common/src/main/java/org/apache/bookkeeper/common/conf/ComponentConfiguration.java [hunk 1]
```diff
@@ -287,4 +291,31 @@ public String getString(String key, String defaultValue) {
...
         return conf.getList(getKeyName(key), defaultValue);
     }
+
+    /**
+     * returns the string representation of json format of this config.
+     *
+     * @return
+     * @throws ParseJsonException
+     */
+    public String asJson() {
```
Comment (reddycharan) [inline bookkeeper-common/src/main/java/org/apache/bookkeeper/common/conf/ComponentConfiguration.java]: but where is this used?

Comment (sijie) [reply bookkeeper-common/src/main/java/org/apache/bookkeeper/common/conf/ComponentConfiguration.java]: it is used in a subsequent pull request: https://github.com/apache/bookkeeper/pull/1553/files#diff-3f4d443c559ae8b071b0229fbb91df2aR209

I am breaking the changes into small changes, so they are easier to be reviewed.

**Comment by eolivelli** [comment]
rebuild java8
rerun pr vaidation

**Comment by sijie** [comment]
rerun pr validation

**Comment by sijie** [comment]
I merged #1553 prior to #1552 . so this change is already included as part of #1553 . so the merge script doesn't work. so I manually merge this through github button. this is to leave a git commit there about the change.

**Pull request merged**